libcloud/compute/drivers/ecs.py

```diff
@@ -1289,6 +1289,102 @@ def create_public_ip(self, instance_id):
             return findtext(resp.object, 'IpAddress',
                             namespace=self.namespace)
 
+    def ex_security_group_ingress_permission(self, group_id,
+                                             ip_protocol, portrange,
+                                             source_cidrip=None):
+        """
+        Author security group access permission(ingress).
+
+        :keyword group_id: Security group id.
+        :type group_id: ``str``
+
+        :keyword ip_protocol: tcp | udp | icmp | gre | all.
+        :type ip_protocol: ``str``
+
+        :keyword portrange: If ip_protocol is tcp|udp, the port range is
+                            [1,65535] and the example value format is 1/200.
+                            It means port range[1,200]. If input is 200/1, an
+                            error will raise.
+                            If ip_protocol is icmp|gre|all port, value format
+                            is -1/-1.
+        :type portrange: ``str``
+
+        :keyword source_cidrip: Source ip range(cidr format), default is
+                                0.0.0.0/0(meaning unlimited). only surport
+                                IPV4.
+        :type source_cidrip: ``str``
+
+        :return: Status
+        :rtype: ``bool``
+        """
+        params = {'Action': 'AuthorizeSecurityGroup',
+                  'RegionId': self.region}
+
+        if group_id is None:
+            raise AttributeError('group_id is required')
+        if ip_protocol is None:
+            raise AttributeError('ip_protocol is required')
+        if portrange is None:
+            raise AttributeError('portrange is required')
+        if source_cidrip is None:
+            source_cidrip = '0.0.0.0/0'
```

Comment:
Instead of doing that, please just set the default value of the argument in the method signature - `def (...., source_cidr_ip='0.0.0.0/0')`